File: frontend/src/lib/types/work-orders.ts
export interface Location {
	id: string;
	address?: string;
	city?: string;
	state_province?: string;
	postal_code?: string;
	country?: string;
	latitude?: number;
	longitude?: number;
}

export interface WorkOrder {
	id: string;
	title: string;
	description?: string;
	status: "Open" | "In Progress" | "Completed" | "Cancelled" | "On Hold";
	priority: "Low" | "Medium" | "High";
	location?: Location;
	location_id?: string;
	assigned_to_user_id?: string;
	created_by_user_id: string;
	created_at: string;
	updated_at: string;
}

export interface WorkOrdersResponse {
	data: WorkOrder[];
	count: number;
	pagination?: {
		page: number;
		limit: number;
		total: number;
		totalPages: number;
	};
}

export interface WorkOrderCreate {
	title: string;
	description?: string;
	status: "Open" | "In Progress" | "On Hold";
	priority: "Low" | "Medium" | "High";
	location_id?: string;
	assigned_to_user_id?: string;
}

export interface WorkOrderUpdate {
	title?: string;
	description?: string;
	status?: "Open" | "In Progress" | "Completed" | "Cancelled" | "On Hold";
	priority?: "Low" | "Medium" | "High";
	location_id?: string;
	assigned_to_user_id?: string;
}

File: backend/app/models/work_orders.py
from enum import Enum

from pydantic import BaseModel


class WorkOrderStatus(str, Enum):
    OPEN = "Open"
    IN_PROGRESS = "In Progress"
    COMPLETED = "Completed"
    CANCELLED = "Cancelled"
    ON_HOLD = "On Hold"


class WorkOrderPriority(str, Enum):
    LOW = "Low"
    MEDIUM = "Medium"
    HIGH = "High"


class Location(BaseModel):
    id: str  # UUID as string
    name: str
    address: str | None = None
    city: str | None = None
    state_province: str | None = None
    postal_code: str | None = None
    country: str | None = None
    latitude: float | None = None
    longitude: float | None = None


class WorkOrder(BaseModel):
    id: str  # UUID as string
    title: str
    description: str | None = None
    status: WorkOrderStatus
    priority: WorkOrderPriority
    location: Location | None = None
    location_id: str | None = None  # UUID as string for location reference
    assigned_to_user_id: str | None = None
    created_by_user_id: str
    created_at: str
    updated_at: str


class WorkOrderCreate(BaseModel):
    title: str
    description: str | None = None
    status: WorkOrderStatus = WorkOrderStatus.OPEN
    priority: WorkOrderPriority = WorkOrderPriority.MEDIUM
    location_id: str | None = None  # UUID as string
    assigned_to_user_id: str | None = None


class WorkOrderUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    status: WorkOrderStatus | None = None
    priority: WorkOrderPriority | None = None
    location_id: str | None = None  # UUID as string
    assigned_to_user_id: str | None = None


class PaginationInfo(BaseModel):
    page: int
    limit: int
    total: int
    totalPages: int


class WorkOrdersResponse(BaseModel):
    data: list[WorkOrder]
    count: int
    pagination: PaginationInfo | None = None
